src/Compilers/Core/MSBuildTask/MvidReader.cs

```diff
@@ -8,56 +8,78 @@ namespace Microsoft.CodeAnalysis.BuildTasks
 {
     public static class MvidReader
     {
+        private static readonly Guid s_empty = Guid.Empty;
+
         public static Guid ReadAssemblyMvidOrEmpty(Stream stream)
         {
             return ReadAssemblyMvidOrEmpty(new BinaryReader(stream));
         }
 
         private static Guid ReadAssemblyMvidOrEmpty(BinaryReader reader)
         {
-            Guid empty = Guid.Empty;
-
-            // DOS Header (64), DOS Stub (64), PE Signature (4), COFF Header (20), PE Header (224), 1x Section Header (40)
-            if (reader.BaseStream.Length < 64 + 64 + 4 + 20 + 224 + 40)
-            {
-                return empty;
-            }
-
             // DOS Header: Magic number (2)
-            if (reader.ReadUInt16() != 0x5a4d) // "MZ"
+            if (!ReadUInt16(reader, out ushort magicNumber) || magicNumber != 0x5a4d) // "MZ"
             {
-                return empty;
+                return s_empty;
             }
 
             // DOS Header: Address of PE Signature (at 0x3C)
-            MoveTo(0x3C, reader);
-            uint lfanew = reader.ReadUInt32();
+            if (!MoveTo(0x3C, reader))
+            {
+                return s_empty;
+            }
+            if (!ReadUInt32(reader, out uint lfanew))
```

Comment:
That's what it's called in the spec, which I agree is not very informative. I'll fix.